### PR TITLE
[BUGFIX] Improve parsing of .ge_store_backend_id

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Use pyparsing to parse `.ge_store_backend_id` which ignores whitespace
- Add newline to newly created `.ge_store_backend_id` files to not conflict with common linting rules
- Comment out failing GCS test until we can mock it appropriately

See also comment here: https://github.com/great-expectations/great_expectations/pull/2896#issuecomment-865477399

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/data_context/store/test_store_backends.py [hunk 1]
```diff
@@ -195,47 +195,47 @@ def test_StoreBackend_id_initialization(tmp_path_factory):
     )
 
     # TODO: Fix GCS Testing
-    # TupleGCSStoreBackend
```
Comment (jcampbell) [inline tests/data_context/store/test_store_backends.py:198]: Are you intending to have this remain commented-out?

Comment (anthonyburdi) [reply tests/data_context/store/test_store_backends.py:198]: I don't feel great about it. Let me take another crack at this to see if I can get a reasonable TupleGCSStoreBackend test.

Comment (anthonyburdi) [reply tests/data_context/store/test_store_backends.py:198]: I couldn't come up with a test that didn't mock so much of GCS that we would essentially be doing an "assert value_i_set == value_i_set". In the future perhaps we can implement a docker GCS emulator like https://github.com/fsouza/fake-gcs-server But I'm happy to discuss if there is a better way in the near term.

File: tests/data_context/store/test_store_backends.py [hunk 2]
```diff
@@ -194,48 +194,27 @@ def test_StoreBackend_id_initialization(tmp_path_factory):
...
 
-    # TODO: Fix GCS Testing
     # TupleGCSStoreBackend
-    # Initialize without store_backend_id and check that it is generated correctly
+    # TODO: Improve GCS Testing e.g. using a docker service to mock
+    # Note: Currently there is not a great way to mock GCS so here we are just testing that a config
+    # with unreachable bucket returns the error store backend id
+    # If we were to mock GCS, we would need to provide the value returned from the TupleGCSStoreBackend which
+    # is circumventing actually testing the store backend.
+
```
Comment (Shinnnyshinshin) [inline tests/data_context/store/test_store_backends.py:203]: This note really helps :) thank you

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM ❤️

**Pull request merged**